forte/data/readers/largemovie_reader.py

```diff
@@ -0,0 +1,116 @@
+  author    = {Maas, Andrew L.  and  Daly, Raymond E.  and  Pham, Peter T.  and  Huang, Dan  and  Ng, Andrew Y.  and  Potts, Christopher},
+  title     = {Learning Word Vectors for Sentiment Analysis},
+  booktitle = {Proceedings of the 49th Annual Meeting of the Association for Computational Linguistics: Human Language Technologies},
+  month     = {June},
+  year      = {2011},
+  address   = {Portland, Oregon, USA},
+  publisher = {Association for Computational Linguistics},
+  pages     = {142--150},
+  url       = {http://www.aclweb.org/anthology/P11-1015}
+}
+"""
+
+import os
+import logging
+import re
+from typing import Iterator
+
+from forte.data.data_pack import DataPack
+from forte.data.data_utils_io import dataset_path_iterator
+from forte.data.readers.base_reader import PackReader
+from ft.onto.base_ontology import Document, Sentence
+
+
+__all__ = [
+    "LargeMovieReader"
+]
+
+
+class LargeMovieReader(PackReader):
+    r""":class:`LargeMovieReader` is designed to read in the Large Movie Review Dataset v1.0.
+    There are two directories [pos/, neg/] for the reviews.
+    Within these directories, reviews are stored in text files named
+    following the convention [[id]_[rating].txt].
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.REPLACE_NO_SPACE = re.compile("(\:)|(\')|(\,)|(\")|(\()|(\))|(\[)|(\])")
+        self.REPLACE_WITH_NEWLINE = re.compile("(<br\s*/><br\s*/>)|(\-)|(\/)|(\.)|(\;)|(\!)|(\?)")
+
+    def preprocess_reviews(self, para):
+        para = self.REPLACE_NO_SPACE.sub("", para.lower())
+        para = self.REPLACE_WITH_NEWLINE.sub("\n", para)
+        return para
+
+    def _collect(self, movie_directory) -> Iterator[str]:
+        r"""Iterator over text files in the data_source.
+
+        Args:
+            movie_directory: directory to the review files.
+
+        Returns: Iterator over files in the path with txt extensions.
+        """
+        logging.info("Reading .txt from %s", movie_directory)
+        return dataset_path_iterator(movie_directory, "txt")
+
+    def _parse_pack(self, file_path: str) -> Iterator[DataPack]:
+        data_pack: DataPack = self.new_pack()
+
+        sent_begin: int = 0
+        doc_text: str = ""
+
+        with open(file_path, encoding="utf8") as doc:
+            for para in doc:
+                para = self.preprocess_reviews(para)
+                sents = para.split("\n")
+                for sent in sents:
+                    if len(sent)>0:
+                        sent = sent.strip()
+                        doc_text += sent + " "
+                        doc_offset = sent_begin + len(sent) + 1
+                        # add sentences.
+                        Sentence(data_pack, sent_begin, doc_offset - 1)
+                        sent_begin = doc_offset
+
+        pos_dir = os.path.basename(os.path.dirname(file_path))
+        movie_file = os.path.basename(file_path)
+        docid, score = movie_file.split('_')
```

Comment:
Let's annotate a few more data types here.

And after annotating the type, you will find that re-assigning different data type to the same variable `score` is not suggested.